### ISSUE TITLE
Resynchronizing UI by client's request

### ISSUE THREAD

**Issue by jonasrotilli** [vaadin/flow#14232]
### Description of the bug

I am often getting the message in the log of Resynchronizing UI by client's request.

The full message is:
Resynchronizing UI by client's request. The network message was lost before reaching the client and the client is reloading the full UI state. This typically happens because of a bad network connection with packet loss or because of some part of the network infrastructure (load balancer, proxy) terminating a push (websocket or long-polling) connection. If you are using push with a proxy, make sure the push timeout is set to be smaller than the proxy connection timeout

It's not connection, apparently it's session related.

I have on the same linux server running another Vaadin 8 application.
They have different names, different ports, in different folders and are NGINX-mapped with different subdomains.

I have already evaluated the other problems related to the topic:
#12640
There was no clear solution, someone raised some possibilities:
- Browser, it's not in my case, it wouldn't happen so often.
- HTTP proxy, I use NGINX for the subdomains, but I use other services like NODE, static HTML and never had a problem. NGINX was configured by default, without any additional configuration, it just throws the subdomain to port X.
- Possibility of mixing sessions: it makes no sense, since I have very little load, it happens even with only 1 user logged in.

#12173
In this problem the user uses long duration push. Not my case, I use simple, default @Push.

#11645
In this problem as I understand it was the slow connection. It's not my case, everything is flying here.

#12173
In this problem the user uses long duration push. Not my case, I use simple, default @Push.

#10096
This is a very similar scenario. But there was no conclusion, the user closed without informing how the issue was resolved and if it was resolved.

#9399
This one he solved by changing the server, it is difficult to assess what the problem was

Anyway, this problem is quite recurrent and should be better explained in the documentation. I downloaded the example available from the site, nothing out of the ordinary, little or no extra configuration.

### Expected behavior

The expectation is that it doesn't lock the user's screen, it's terrible to have to ask him to refresh the page, because after it's broken it doesn't come back.

### Minimal reproducible example

It's hard to simulate, because it doesn't always happen.
The impression is that it happens after a while without changes on the page, but sometimes it happens right after logging in or during some slower operation.

### Versions

- Vaadin / Flow version: 23.2.0.alpha1
- Java version: 
openjdk version "11.0.3" 2019-04-16
OpenJDK Runtime Environment (build 11.0.3+7-Ubuntu-1ubuntu218.10.1)
OpenJDK 64-Bit Server VM (build 11.0.3+7-Ubuntu-1ubuntu218.10.1, mixed mode, sharing)

- OS version:
-Ubuntu

- Browser version (if applicable): 
Chrome

**Comment by jonasrotilli** [comment]
Difficult.
I increased the timeout on NGINX to 600 seconds and the problem continues.

**Comment by tiagomartins91** [comment]
Same on Vaadin version 14.8.14

WARN com.vaadin.flow.server.communication.ServerRpcHandler [http-nio-8080-exec-3] Resynchronizing UI by client's request. Under normal operations this should not happen and may indicate a bug in Vaadin platform. If you see this message regularly please open a bug report at https://github.com/vaadin/flow/issues

**Comment by jonasrotilli** [comment]
> tiagomartins91

Apparently no one from Vaadin is watching here..
Let's try to find the problem ourselves, try to see what we have in common.
1 - Do you have any other Vaadin application running on the same server?
A: I do, but it's another folder, another version, nothing shared.

2 - Does it happen in development, when running with Intelij or similar?
A: No.

3 - Does it happen in production?
A: Yes, I stop the service and put a new version, first login most of the time happens, which disproves the theory that it's because of time without moving.

**Comment by jonasrotilli** [comment]
I was hoping to be a conflict between two applications.
But it is not. I completely stopped the other application and started the new one, Vaadin 23.

And the same problem happened:
Resynchronizing UI by client's request. The network message was lost before reaching the client and the client is reloading the full UI state. This typically happens because of a bad network connection with packet loss or because of some part of the network infrastructure (load balancer, proxy) terminating a push (websocket or long-polling) connection. If you are using push with a proxy, make sure the push timeout is set to be smaller than the proxy connection timeout

This is very bad, it is a very serious problem.

**Comment by knoobie** [comment]
I would suggest to post your nginx and push configuration.

**Comment by jonasrotilli** [comment]
> Sugiro postar sua configuração nginx e push.

Push detault:

```
@Theme(value = "myapp")
@PWA(name = "upCampo", shortName = "upCampo")
@NpmPackage(value = "line-awesome", version = "1.3.0")
@Push
```

NGINX file:

```
server {

    server_name  novoportal.MYWEBSITE;

    location / {
        proxy_pass  http://127.0.0.1:1628;
    }

    listen [::]:443 ssl; # managed by Certbot
    listen 443 ssl; # managed by Certbot
    ssl_certificate /etc/letsencrypt/live/novoportal.MYWEBSITE/fullchain.pem; # managed by Certbot
    ssl_certificate_key /etc/letsencrypt/live/novoportal.MYWEBSITE/privkey.pem; # managed by Certbot
    include /etc/letsencrypt/options-ssl-nginx.conf; # managed by Certbot
    ssl_dhparam /etc/letsencrypt/ssl-dhparams.pem; # managed by Certbot


}
server {
    if ($host = novoportal.MYWEBSITE) {
        return 301 https://$host$request_uri;
    } # managed by Certbot


    listen 80;
    listen [::]:80;

    server_name  novoportal.MYWEBSITE;
    return 404; # managed by Certbot

   proxy_read_timeout 600;
   proxy_connect_timeout 600;
   proxy_send_timeout 600;


}
```


About the timeouts in NGINX, I added more time, it didn't make any difference with or without this part of the code:

```

   proxy_read_timeout 600;
   proxy_connect_timeout 600;
   proxy_send_timeout 600;
```

**Comment by knoobie** [comment]
I don't see anything related to push in the configuration. Cuba has a example for nginx that you could try: https://doc.cuba-platform.com/manual-latest/server_push_settings.html#server_push_settings_using_proxy - important is the part about `upgrade`

I'm more experienced with apache httpd and there it is a must have to configure websockets corretly to work in corporate networks.

**Comment by jonasrotilli** [comment]
> Não vejo nada relacionado ao push na configuração. Cuba tem um exemplo para nginx que você pode tentar: https://doc.cuba-platform.com/manual-latest/server_push_settings.html#server_push_settings_using_proxy - importante é a parte sobre`upgrade`
> 
> Eu sou mais experiente com apache httpd e aí é necessário configurar websockets corretamente para trabalhar em redes corporativas.

I added your suggestion in NGINX.
Good news: so far, the problem hasn't happened yet.
I'll leave it running during the day and come back here to confirm if it worked or not.

Thank you very much!

**Comment by jonasrotilli** [comment]
> I don't see anything related to push in the configuration. Cuba has a example for nginx that you could try: https://doc.cuba-platform.com/manual-latest/server_push_settings.html#server_push_settings_using_proxy - important is the part about `upgrade`
> 
> I'm more experienced with apache httpd and there it is a must have to configure websockets corretly to work in corporate networks.

Sorted out!
Big help. I accessed the CUBA link and used the "location" part, it seems that there is something related to WebSocket support.
Since putting it on, I haven't had any more problems.
@knoobie  Thank you so much for your help, it saved me from several nights sleep!

I share here my NGINX that I'm using for other users, and I won't close the ticket so that someone from Vaadin can evaluate if any improvement is needed in relation to the theme.
I believe this could be in the documentation.

Here's the complete file:




   ```
server {

 server_name  mywebsite.com;

    location / {
        proxy_set_header X-Forwarded-Host $host;
        proxy_set_header X-Forwarded-Server $host;
        proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for;
        proxy_read_timeout     3600;
        proxy_connect_timeout  240;
        proxy_set_header Host $host;
        proxy_set_header X-RealIP $remote_addr;

        proxy_pass  http://127.0.0.1:PORT_EXIT_DO_YOUR_SPRINGBOOT;

        proxy_set_header X-Forwarded-Proto $scheme;

        proxy_set_header Upgrade $http_upgrade;
        proxy_set_header Connection "upgrade";
}

    listen [::]:443 ssl; # managed by Certbot (Certificate SSL)
    listen 443 ssl; # managed by Certbot (Certificate SSL)
    ssl_certificate /etc/letsencrypt/live/mywebsite.com/fullchain.pem; # managed by Certbot (Certificate SSL)
    ssl_certificate_key /etc/letsencrypt/live/mywebsite.com/privkey.pem; # managed by Certbot (Certificate SSL)
    include /etc/letsencrypt/options-ssl-nginx.conf; # managed by Certbot (Certificate SSL)
    ssl_dhparam /etc/letsencrypt/ssl-dhparams.pem; # managed by Certbot (Certificate SSL)


}
server {
    if ($host = mywebsite.com) {
        return 301 https://$host$request_uri;
    } # managed by Certbot


    listen 80;
    listen [::]:80;

    server_name  mywebsite.com;
    return 404; # managed by Certbot (Certificate SSL)

   proxy_read_timeout 600;
   proxy_connect_timeout 600;
   proxy_send_timeout 600;

}
```

**Comment by knoobie** [comment]
> I believe this could be in the documentation.

How to configure a reverse proxy should be an important topic inside the docs (cc: @tarekoraby)

**Comment by jonasrotilli** [comment]
I have bad news.
Monitoring since that day and it happened again.
Less often than before the NGINX change but it is happening.
Any other possible problems?

**Comment by tiagomartins91** [comment]
Same here

**Comment by jonasrotilli** [comment]
That is terrible. We never had problems like this in the old Vaadin 8 application.
Imagine the inconvenience this is causing the customer.
This image makes me shiver every time I see it:

<img width="1435" alt="Captura de Tela 2022-08-09 às 09 29 42" src="https://user-images.githubusercontent.com/8934198/183660490-c8da01f6-d065-4b1b-b639-c44c70a6ecd8.png">


Please someone help us!

**Comment by tarekoraby** [comment]
I would personally try to obtain a debug log of nginx to try to understand what's happening.

**Comment by jonasrotilli** [comment]
> Eu pessoalmente tentaria obter um log de depuração do nginx para tentar entender o que está acontecendo.

I am fully available to collect the data needed to resolve this issue.

Help me, how do I do this?

**Comment by tarekoraby** [comment]
I'm not an nginx expert, but I'd check the docs for instructions on that: https://nginx.org/en/docs/debugging_log.html.

**Comment by tiagomartins91** [comment]
Any update? Still happens sometimes and needs to refresh the page for the application work

**Comment by mshabarov** [comment]
We are going to investigate it more closely in the upcoming development iteration.

**Comment by kagian** [comment]
I've managed to reproduce this issue almost consistently. There are a couple of interesting factors that lead into this. I have a pcap and a google chrome debug output and the code that generates the issue.



```
java.lang.UnsupportedOperationException: Unexpected message id from the client. Expected sync id: 9, got 10. more details logged on DEBUG level.```

**Comment by knoobie** [comment]
@kagian Please share it, if you wanna see steady progress in this topic.

**Comment by flefebure** [comment]
Our multi-years Vaadin 7->14 migration work just went in production.

During dev/tests period, we had occasionnally this problem, principally when remote working, with a bad network.
But now we have more users, and users with various network configuration, we realize that the problem is more serious than expected.
We speak a lot of NGinx in this thread, my feeling is that this problem is not reverse-proxy related.
For sure we have an NGinx in front of the app, but we also access directly the tomcat port, and we can see the problem at the same frequency.
My short term goal is to produce a small project that reproduces the problem, for the Vaadin team.
I'll probably use one of those browser extensions that simulates a bad network.

This problem is very critical, and as someone said, it wasn't observable with the old 7/8 vaadin platform

**Comment by TatuLund** [comment]
> We speak a lot of NGinx in this thread, my feeling is that this problem is not reverse-proxy related.

@flefebure, @kagian, @jonasrotilli  The problem can be reverse-proxy related, but there are number of other possible causes as well. E.g. slow VPN, flaky Wifi / cellular network etc.

Also framework corner case bugs are a possibility, it is not long ago we introduced this fix, so I recommend to use the latest Vaadin 14 or 23 version, and observe if is more stabble in your environment. Just pointing this out as the ticket was reported before this fixx was introduced and no comment here mentions which specific Vaadin version is in use.

https://github.com/vaadin/flow/pull/13733

**Comment by kagian** [comment]
Just to clarify here, the protocol in use seemed to be TCP as far as I could see. There is also no proxy or additional rewrite components in the path. So for an end to end TCP session, there really should not be the possibility of loss without recovery. And as such it should not be possible to get an out of sequence or dropped packets (unless this really is in UDP, which didn't seem like it).

It really does look like some kind of state tracking issue in Vaading server side. I'll check in on the new version, however, the current change to moving over to manual push updates seems to work well and as a result has reduced the want to change this again on our side.

**Comment by flefebure** [comment]
@tatulund we just upgraded Vaadin 14.8.4->14.8.17 et Flow 2.7.11->2.7.20 
Now we wait for users feedback
[cross fingers]

**Comment by jonasrotilli** [comment]
To complete:
It decreased a lot after I made the suggestions for changes in NGINX that I posted there at the beginning.
But it didn't completely solve it.
It started to happen more often. No changes to the version of either Vaadin or any changes to the server in question.
I believe this needs further investigation.

**Comment by jonasrotilli** [comment]
I just updated to version 23.1.10 and the problem continues.
See log image in attachment.
![Captura de Tela 2022-09-21 às 08 51 28](https://user-images.githubusercontent.com/8934198/191508657-c50bba91-ff1a-428c-91a4-641980602f0b.png)

**Comment by tiagomartins91** [comment]
Any update about this?

It happens more often with the last release. It's impossible to deliver or update a vaadin application in production with this issue.

I have another application running vaadin 8 in production and this doesn't happen. Use the same NGINX reverse proxy and the same configuration, without any problem.

**Comment by mshabarov** [comment]
Message from @jonasrotilli which is not shown here for some reason:

> I just updated to version 23.1.10 and the problem continues.
See log image in attachment.
![image](https://user-images.githubusercontent.com/61410877/193270932-fe592ef2-b67a-4c11-9e4e-6dfaa89338ad.png)

**Comment by echarlus** [comment]
> In my case, just stop the refresh lopping page if I force it to refresh.

yes this works most of the time. But asking a customer to do that is not an option ... I also encountered cases where I had to clear all the stored data (session etc) and restart navigation before I could reach the site again :(

**Comment by echarlus** [comment]
> Any update about this?
> 
> It happens more often with the last release. It's impossible to deliver or update a vaadin application in production with this issue.
> 
> I have another application running vaadin 8 in production and this doesn't happen. Use the same NGINX reverse proxy and the same configuration, without any problem.

Agreed I never experienced this issue with Vaadin 8, with 23 it's happening very often and my customers are becoming angry. I hope the fix will be delivered quickly

**Comment by caalador** [comment]
Is there a chance to get client logs of when this happens?
I would hope for some console warns and logs on client and server ids.

**Comment by echarlus** [comment]
> #13733

It's happening on production so client-side logging is minimal. I have the chrome dev console opened and if I manage to catch something I'll post it here.

**Comment by echarlus** [comment]
I'm getting this "violation" logs from time to time, since one of them mentions "readystatechange" I'm posting them here just in case : `
[Violation] 'readystatechange' handler took 356ms 
[Violation] Forced reflow while executing JavaScript took 152ms
generated-flow-imports.2848e01a.js:6550 [Violation] 'setTimeout' handler took 51ms
FlowBootstrap.0b77bed3.js:1 [Violation] 'setTimeout' handler took 54ms
`
![image](https://user-images.githubusercontent.com/7509875/193761661-8c7c94df-817d-45fb-9b0e-21cdd64a8395.png)

**Comment by caalador** [comment]
I was hoping for `Received resync message`, `Forced update of clientId ...`, `Updating client-to-server id ...` or `Server expects next client-to-server ...` when the issue happens

**Comment by echarlus** [comment]
@caalador I've just experienced a weird behavior that I had already seen but this time I was able to gather some info. Not sure it's the same issue as the one we're dealing with here but I'm posting the logs anyway just in case. 
The client made a request to the server, the request took a long time to process and the Azure AppGateway sitting in between the client & server did reset the connection (timeout is set at 60s) it therefore replied with a 502 HTTP code.
On the client side, I then started to see repeated attempts to query the sever but the chrome console continued displaying the 502 error, however on my server I saw no incoming requests (Tomcat logs do not show any external request during this time). After some times, I finally saw a "online" message displayed on the client and, on the server I saw a flow of requests with all led to the error `INFO com.vaadin.flow.server.communication.ServerRpcHandler - Ignoring old duplicate message from the client. Expected: 14, got: 13`
Here's a screenshot of the chrome console and the corresponding part of both catalina.out and access.log of tomcat.

![image](https://user-images.githubusercontent.com/7509875/193767723-92e48093-185b-42f3-b8db-7e2b47e10be0.png)
catalina.out
`
[http-nio-8080-exec-3] INFO com.vaadin.flow.server.communication.ServerRpcHandler - Ignoring old duplicate message from the client. Expected: 14, got: 13
[http-nio-8080-exec-2] INFO com.vaadin.flow.server.communication.ServerRpcHandler - Ignoring old duplicate message from the client. Expected: 14, got: 13
[http-nio-8080-exec-4] INFO com.vaadin.flow.server.communication.ServerRpcHandler - Ignoring old duplicate message from the client. Expected: 14, got: 13
[http-nio-8080-exec-8] INFO com.vaadin.flow.server.communication.ServerRpcHandler - Ignoring old duplicate message from the client. Expected: 14, got: 13
[http-nio-8080-exec-5] INFO com.vaadin.flow.server.communication.ServerRpcHandler - Ignoring old duplicate message from the client. Expected: 14, got: 13
[http-nio-8080-exec-14] INFO com.vaadin.flow.server.communication.ServerRpcHandler - Ignoring old duplicate message from the client. Expected: 14, got: 13
[http-nio-8080-exec-15] INFO com.vaadin.flow.server.communication.ServerRpcHandler - Ignoring old duplicate message from the client. Expected: 14, got: 13
[http-nio-8080-exec-6] INFO com.vaadin.flow.server.communication.ServerRpcHandler - Ignoring old duplicate message from the client. Expected: 14, got: 13
[http-nio-8080-exec-7] INFO com.vaadin.flow.server.communication.ServerRpcHandler - Ignoring old duplicate message from the client. Expected: 14, got: 13
[http-nio-8080-exec-9] INFO com.vaadin.flow.server.communication.ServerRpcHandler - Ignoring old duplicate message from the client. Expected: 14, got: 13
[http-nio-8080-exec-11] INFO com.vaadin.flow.server.communication.ServerRpcHandler - Ignoring old duplicate message from the client. Expected: 14, got: 13
[http-nio-8080-exec-3] INFO com.vaadin.flow.server.communication.ServerRpcHandler - Ignoring old duplicate message from the client. Expected: 15, got: 14
[http-nio-8080-exec-4] INFO com.vaadin.flow.server.communication.ServerRpcHandler - Ignoring old duplicate message from the client. Expected: 15, got: 14
[http-nio-8080-exec-5] INFO com.vaadin.flow.server.communication.ServerRpcHandler - Ignoring old duplicate message from the client. Expected: 15, got: 14
[http-nio-8080-exec-3] WARN com.vaadin.flow.server.communication.ServerRpcHandler - Resynchronizing UI by client's request. A network message was lost before reaching the client and the client is reloading the full UI state. This typically happens because of a bad network connection with packet loss or because of some part of the network infrastructure (load balancer, proxy) terminating a push (websocket or long-polling) connection. If you are using push with a proxy, make sure the push timeout is set to be smaller than the proxy connection timeout
`

access.log -> here you see a hole between 9:35 and 9:40 then until 9:53 (this corresponds to the period during which the client requests where showing 502 responses but the server did not receive anything)
 `10.1.0.5 - - [04/Oct/2022:09:35:15 +0200] "POST /?v-r=uidl&v-uiId=1 HTTP/1.1" 200 35
10.1.0.5 - - [04/Oct/2022:09:35:36 +0200] "POST /?v-r=uidl&v-uiId=1 HTTP/1.1" 200 64956
10.1.0.5 - - [04/Oct/2022:09:35:36 +0200] "POST /?v-r=uidl&v-uiId=1 HTTP/1.1" 200 131
10.1.0.5 - - [04/Oct/2022:09:35:36 +0200] "POST /?v-r=uidl&v-uiId=1 HTTP/1.1" 200 132
10.1.0.5 - - [04/Oct/2022:09:35:36 +0200] "POST /?v-r=uidl&v-uiId=1 HTTP/1.1" 200 14712
10.1.0.5 - - [04/Oct/2022:09:35:36 +0200] "POST /?v-r=uidl&v-uiId=1 HTTP/1.1" 200 132
10.1.0.4 - - [04/Oct/2022:09:39:48 +0200] "POST /?v-r=heartbeat&v-uiId=1 HTTP/1.1" 200 -
10.1.0.4 - - [04/Oct/2022:09:40:27 +0200] "POST /?v-r=uidl&v-uiId=1 HTTP/1.1" 200 37
10.1.0.4 - - [04/Oct/2022:09:53:07 +0200] "POST /?v-r=uidl&v-uiId=1 HTTP/1.1" 200 37
10.1.0.4 - - [04/Oct/2022:09:53:07 +0200] "POST /?v-r=uidl&v-uiId=1 HTTP/1.1" 200 37
10.1.0.4 - - [04/Oct/2022:09:53:07 +0200] "POST /?v-r=uidl&v-uiId=1 HTTP/1.1" 200 37
10.1.0.5 - - [04/Oct/2022:09:53:07 +0200] "POST /?v-r=heartbeat&v-uiId=1 HTTP/1.1" 200 -
10.1.0.4 - - [04/Oct/2022:09:53:07 +0200] "POST /?v-r=uidl&v-uiId=1 HTTP/1.1" 200 37
10.1.0.4 - - [04/Oct/2022:09:53:07 +0200] "POST /?v-r=uidl&v-uiId=1 HTTP/1.1" 200 37
10.1.0.5 - - [04/Oct/2022:09:53:07 +0200] "POST /?v-r=heartbeat&v-uiId=1 HTTP/1.1" 200 -
10.1.0.4 - - [04/Oct/2022:09:53:07 +0200] "POST /?v-r=uidl&v-uiId=1 HTTP/1.1" 200 37
10.1.0.4 - - [04/Oct/2022:09:53:07 +0200] "POST /?v-r=uidl&v-uiId=1 HTTP/1.1" 200 37
10.1.0.4 - - [04/Oct/2022:09:53:07 +0200] "POST /?v-r=uidl&v-uiId=1 HTTP/1.1" 200 37
10.1.0.4 - - [04/Oct/2022:09:53:07 +0200] "POST /?v-r=uidl&v-uiId=1 HTTP/1.1" 200 -
10.1.0.4 - - [04/Oct/2022:09:53:07 +0200] "POST /?v-r=uidl&v-uiId=1 HTTP/1.1" 200 37
10.1.0.4 - - [04/Oct/2022:09:53:07 +0200] "POST /?v-r=uidl&v-uiId=1 HTTP/1.1" 200 37
10.1.0.4 - - [04/Oct/2022:09:53:07 +0200] "POST /?v-r=uidl&v-uiId=1 HTTP/1.1" 200 37
`
Let me know if this helps.

**Comment by echarlus** [comment]
@caalador while playing with queries & trying to created issues I managed to get the following behavior : 

- Page takes a long time to display (as above)
- Servers replies with 502
- This time I got the message "connection lost trying to reconnect"
- On the server I saw that after a while I got the exception caused by the proxy closing the connection
- On the client it started to replay the same query which finally succeeded 
- The the page started to redraw indefinitely (but no query sent to server) and I got the following exception in the console : 

`FlowClient.947c8d40.js:1 Uncaught (in promise) Error: Client is resynchronizing
    at FlowClient.947c8d40.js:1:42744
    at Array.forEach (<anonymous>)
    at Zy (FlowClient.947c8d40.js:1:42719)
    at by (FlowClient.947c8d40.js:1:34159)
    at z.db (FlowClient.947c8d40.js:3:76660)
    at x (FlowClient.947c8d40.js:1:32329)
    at Map.forEach (<anonymous>)
    at i2 (FlowClient.947c8d40.js:1:43877)
    at W1 (FlowClient.947c8d40.js:3:50372)
    at mf (FlowClient.947c8d40.js:3:37591)
    at z.Bb (FlowClient.947c8d40.js:3:74625)
    at z.O (FlowClient.947c8d40.js:3:86799)
    at XMLHttpRequest.<anonymous> (FlowClient.947c8d40.js:1:25882)
    at od (FlowClient.947c8d40.js:1:16760)
    at Z2 (FlowClient.947c8d40.js:3:5692)
    at XMLHttpRequest.<anonymous> (FlowClient.947c8d40.js:1:28235)`

**Comment by Artur-** [comment]
Is there a way to reproduce this with e.g. an nginx proxy with a timeout and a simple Vaadin app that just sleeps in a click listener?

**Comment by echarlus** [comment]
I have a setting with a Microsoft Azure AppGateway that acts as a load balancer and behind that, I have tomcat-9 as an app server. 
I can reproduce by making a request timeout (just sleep on the server side) which causes the app-gateway to close the connection and return a 502 error.

**Comment by echarlus** [comment]
Hi @Artur- did you manage to reproduce ? any update ?

**Comment by caalador** [comment]
Haven't been able to reproduce the issue. All my setups to get it to fail have failed in one way or another.

**Comment by echarlus** [comment]
Is there anything I can do to help ?

**Comment by caalador** [comment]
Not having any proxy setups it took me a while to get a proxy to work.>
If I have the timeout short enough I do get a 504 response without the server being done, but it doesn't send me into a request loop as such only 2 more of the same request are sent before the server is done with the initial thread. So as the client resends the xhr request again for a != 200 response error it will send it to the server until the original is done and the server responds with a 200 response. 

I would expect the amount not sent to the server is just blocked by the proxy as coming too fast?

The application stops responding to any input after that which is the same that happens with #14470 and only a reload fixes the app.

So as a first help I would expect having the long event run in a thread releasing the request immediately and using push (or ui polling) for update after it is finished to fix the biggest problem as waiting over 60s for the server seems excessive. 

I'll look into why the resynched page stops responding.

**Comment by echarlus** [comment]
@caalador thanks for the feedback. I know 60s is far too long for a response and I'm working on running the task in the background and using push. However, I've had many users complaining about the UI not responding to any click at some occasions (and they had not been getting the 60s timeout) so maybe fixing this would solve most of my problem. 
Did you manage to get the JS exception for which I posted a stack trace above ? 
If you need access to my app to do some investigation let me know, I can get you an account.

**Comment by caalador** [comment]
That one I didn't get, but that is probably due to the client having  waiting promises when the resych is requested.
So any pending promises will be rejected on a resychronization.

**Comment by echarlus** [comment]
@caalador I encountered the issue again this morning. Got a 502 on one page, then the whole site stopped responding. 
I tried to reload the page or go to the home page, nothing happened. On the server side I have to logs of any request coming in ...
I see in the network view that the requests are being handled by sw.js and they all fail now. Could this be that sw.js is causing the issue (maybe that's what you're already looking at ...)
![image](https://user-images.githubusercontent.com/7509875/194260518-3f9f7266-5f02-46e4-9297-15bffb5e3d71.png)

**Comment by echarlus** [comment]
I can confirm that sw.js seems to be causing the whole behavior (even blocking the home page display) after it received a first 502 response. 
In chrome I stopped sw.js and then restarted it and everything went back to normal ...
![image](https://user-images.githubusercontent.com/7509875/194261447-5208c9c0-e84b-4a8a-93bc-05d93253b30d.png)

**Comment by caalador** [comment]
Found one issue that made it so that not all changes for the resync were executed if there was a missing node.
For instance the notification in 14470 disappearing during the resync operation being the simplest reproducible one.

**Comment by echarlus** [comment]
> Found one issue that made it so that not all changes for the resync were executed if there was a missing node. For instance the notification in 14470 disappearing during the resync operation being the simplest reproducible one.

great : ) @caalador is there a way I can get a version with the fix to test it on my end ?

**Comment by caalador** [comment]
You can grab a snapshot a while after the fix has merged using the pre-releases repository.
```
        <repository>
            <id>vaadin-prereleases</id>
            <url>
                https://maven.vaadin.com/vaadin-prereleases/
            </url>
        </repository>
```
I'll post when I can see a snapshot build for 23.2 containing the fix.

**Comment by echarlus** [comment]
@caalador thanks. 
Does the fix also take care of what I've seen on the client side (sw.js that seems to stop from working and blocking further connections) ?

**Comment by caalador** [comment]
At least I didn't note a problem with service worker, but it might need some old worker there.
If this fix doesn't help lets open up a new ticket for looking into service worker problem.

**Comment by echarlus** [comment]
Can you describe what the fix you released here addresses exactly in terms of behavior (seen by the user) ? My feeling tells me I can still be having other issues that are not addressed here.

**Comment by caalador** [comment]
The application keeps on working after a resync instead of freezing up so that it doesn't take commands anymore.

**Comment by echarlus** [comment]
ok. Does that takes care of the Exception I reported on the client side : 
`FlowClient.947c8d40.js:1 Uncaught (in promise) Error: Client is resynchronizing
    at FlowClient.947c8d40.js:1:42744
    at Array.forEach (<anonymous>)
    at Zy (FlowClient.947c8d40.js:1:42719)
    at by (FlowClient.947c8d40.js:1:34159)
    at z.db (FlowClient.947c8d40.js:3:76660)
    at x (FlowClient.947c8d40.js:1:32329)
    at Map.forEach (<anonymous>)
    at i2 (FlowClient.947c8d40.js:1:43877)
    at W1 (FlowClient.947c8d40.js:3:50372)
    at mf (FlowClient.947c8d40.js:3:37591)
    at z.Bb (FlowClient.947c8d40.js:3:74625)
    at z.O (FlowClient.947c8d40.js:3:86799)
    at XMLHttpRequest.<anonymous> (FlowClient.947c8d40.js:1:25882)
    at od (FlowClient.947c8d40.js:1:16760)
    at Z2 (FlowClient.947c8d40.js:3:5692)
    at XMLHttpRequest.<anonymous> (FlowClient.947c8d40.js:1:28235)`

?

**Comment by caalador** [comment]
Not specifically. Also that is not an error as such as it is noting that the promise is cancelled due to resync:
     * Reject all promises pending on this server object. Called during client
     * resynchronization to free consumers of promises that are never delivered
     * by the server.

**Comment by caalador** [comment]
Now the 23.2-SNAPSHOT should be available.

**Comment by echarlus** [comment]
Thanks, I'll try that and see how it goes !

**Comment by tiagomartins91** [comment]
Wha is the expected time to be released on Vaadin 14.x version?

**Comment by caalador** [comment]
Next release.

**Comment by echarlus** [comment]
And do you know when the next official 23.2 will be released ? Will it automatically include this patch ?

**Comment by mshabarov** [comment]
@echarlus @tiagomartins91 since this is a blocker issue, we'll do Flow release asap. I've already started Flow `23.2.4` release, so it should be available soon in the maven central, so you can override Flow version in your pom.xml and test it. V14.8 needs the back-porting and validation process to be completed, so we'll release flow version for 14.8 most likely on Monday next week. We will release next Vaadin versions (including mentioned flow versions) for 23.2 and 14.8 next week once we merge other important bug-fixes in.

**Comment by echarlus** [comment]
@mshabarov thanks a lot ! I'll pick-it up as soon as it's available.

**Comment by echarlus** [comment]
@mshabarov can you please confirm I should be looking here : https://mvnrepository.com/artifact/com.vaadin/flow-server ? 
Thanks

**Comment by mshabarov** [comment]
@echarlus no, this is the official one https://search.maven.org/artifact/com.vaadin/flow-bom

**Comment by echarlus** [comment]
@mshabarov thanks. I've updated my repo-source URL.

**Comment by echarlus** [comment]
@mshabarov sorry but I'm not a maven pro and I cannot manage to get the flow version 23.2.4  to get used when compiling ...
I've added this in my dependency : 
`<dependencyManagement>
        <dependencies>
            <dependency>
                <groupId>com.vaadin</groupId>
                <artifactId>vaadin-bom</artifactId>
                <type>pom</type>
                <scope>import</scope>
                <version>${vaadin.version}</version>
            </dependency>

            <dependency>
                <groupId>com.vaadin</groupId>
                <artifactId>flow-bom</artifactId>
                <version>23.2.4</version>
                <type>pom</type>
                <scope>import</scope>
            </dependency>
        </dependencies>
    </dependencyManagement>
`
${vaadin.version} is still 23.2.3 since the whole 23.2.4 is not released yet. 
When I build, it's still using 23.2.3, what's the way to make it work ? 
Thanks.

**Comment by mcollovati** [comment]
Try to move the flow-bom dependency before vaadin-bom.
The other dependency you added (com.vaadin:flow) is wrong.
If you remove it and do a mvn dependency:tree you should see Flow-client at version 23.2.4
If not, add a dependency to com.vaadin:flow-client at version 23.2.4

**Comment by echarlus** [comment]
Thanks now it has picked-up 23.2.4.

**Comment by echarlus** [comment]
@caalador I've been testing the new version and unfortunately I'm still experiencing the issue I had previously : 
* Server is too slow to reply -> proxy sends a 502 error
* Client will then be trying to replay the same request and I'm under the impression that the 502 reply is cached locally so it gets again 502 (I have not trace of the query on the server side)
* I see some heartbeat queries coming in on the server and it replies
* After a while I get an "Online message" on the client side 
* Query is issued again (I see it on the server this time) and I also see the message
`[http-nio-8080-exec-14] INFO com.vaadin.flow.server.communication.ServerRpcHandler - Ignoring old duplicate message from the client. Expected: 26, got: 25
[http-nio-8080-exec-14] WARN com.vaadin.flow.server.communication.ServerRpcHandler - Resynchronizing UI by client's request. A network message was lost before reaching the client and the client is reloading the full UI state. This typically happens because of a bad network connection with packet loss or because of some part of the network infrastructure (load balancer, proxy) terminating a push (websocket or long-polling) connection. If you are using push with a proxy, make sure the push timeout is set to be smaller than the proxy connection timeout`
*The query goes fine on the server (my logs indicate that the server replied faster than the proxy timeout) but the page is never displayed on the client
* On the client I see that it's still replaying the same query forever.
* If I try to navigate to another page, the client does not send the request to the server since it's in "offline" mode (Displaying connection lost trying to reconnect, even if the hearbeat messages seem to go through) - logs on the server indicate that they do : 
`10.1.0.4 - - [08/Oct/2022:16:35:28 +0200] "POST /?v-r=uidl&v-uiId=2 HTTP/1.1" 200 -
10.1.0.4 - - [08/Oct/2022:16:35:28 +0200] "POST /?v-r=heartbeat&v-uiId=2 HTTP/1.1" 200 -
10.1.0.5 - - [08/Oct/2022:16:36:16 +0200] "POST /?v-r=heartbeat&v-uiId=0 HTTP/1.1" 200 -
10.1.0.4 - - [08/Oct/2022:16:36:16 +0200] "POST /?v-r=heartbeat&v-uiId=1 HTTP/1.1" 200 -`

I've saved the chrome dev tools logs (console & network activity) as well as server logs. 
I'll send you a link to them privately so that you can check them out if you like.
I'm available if you need more info / debug logs.

**Comment by knoobie** [comment]
Just an idea: if you don't heavily rely on PWA - remove the annotation and deactivate the already locally installed service worker for that domain in your browser and see if that  improves your situation

**Comment by echarlus** [comment]
Good idea, I'll try that ...

**Comment by mshabarov** [comment]
[Flow 2.7.23](https://github.com/vaadin/flow/releases/tag/2.7.23) has been sent to maven-central and should be available there soon.

**Comment by echarlus** [comment]
@caalador I've just experienced the issue again, on a server which did not had any latency issue and did not sent back a 502 response after a timeout. 
I logged in and then I saw the page starting to refresh itself for a while, I captures both the network & console log on chrome, one can see the exception in the logs (full log available upon request): 
`vaadinPush-min.js?v=23.2.4:1 Vaadin push loaded
generated-flow-imports.679d9aff.js:7824 WARNING: Since Vaadin 23.1, using <iron-icon> is deprecated. Please use <vaadin-icon> and '@vaadin/vaadin-lumo-styles/vaadin-iconset.js' instead.
(anonyme) @ generated-flow-imports.679d9aff.js:7824
generated-flow-imports.679d9aff.js:11260 The main 'lit-element' module entrypoint is deprecated. Please update your imports to use the 'lit' package: 'lit' and 'lit/decorators.ts' or import from 'lit-element/lit-element.ts'. See https://lit.dev/msg/deprecated-import-path for more information.
(anonyme) @ generated-flow-imports.679d9aff.js:11260
generated-flow-imports.679d9aff.js:11264 WARNING: Since Vaadin 23.2, "@vaadin/vaadin-ordered-layout" is deprecated. Use "@vaadin/vertical-layout" instead.
(anonyme) @ generated-flow-imports.679d9aff.js:11264
generated-flow-imports.679d9aff.js:11498 WARNING: Since Vaadin 23.2, "@vaadin/vaadin-form-layout" is deprecated. Use "@vaadin/form-layout" instead.
(anonyme) @ generated-flow-imports.679d9aff.js:11498
generated-flow-imports.679d9aff.js:11502 WARNING: Since Vaadin 23.2, "@vaadin/vaadin-ordered-layout" is deprecated. Use "@vaadin/horizontal-layout" instead.
(anonyme) @ generated-flow-imports.679d9aff.js:11502
generated-flow-imports.679d9aff.js:11506 WARNING: Since Vaadin 23.2, "@vaadin/vaadin-button" is deprecated. Use "@vaadin/button" instead.
(anonyme) @ generated-flow-imports.679d9aff.js:11506
vaadinPush-min.js?v=23.2.4:1 Invoking executeWebSocket, using URL: wss://test.o-cell.fr/?v-r=push&v-uiId=0&v-pushId=5b3f1338-8707-4ef9-ba2c-d9055e81b859&X-Atmosphere-tracking-id=0&X-Atmosphere-Framework=3.1.2-javascript&X-Atmosphere-Transport=websocket&X-Atmosphere-TrackMessageSize=true&Content-Type=application/json; charset=UTF-8&X-atmo-protocol=true
vaadinPush-min.js?v=23.2.4:1 WebSocket connection to 'wss://test.o-cell.fr/?v-r=push&v-uiId=0&v-pushId=5b3f1338-8707-4ef9-ba2c-d9055e81b859&X-Atmosphere-tracking-id=0&X-Atmosphere-Framework=3.1.2-javascript&X-Atmosphere-Transport=websocket&X-Atmosphere-TrackMessageSize=true&Content-Type=application/json;%20charset=UTF-8&X-atmo-protocol=true' failed: 
ax @ vaadinPush-min.js?v=23.2.4:1
vaadinPush-min.js?v=23.2.4:1 Mon Oct 10 2022 15:16:01 GMT+0200 (heure d’été d’Europe centrale) Atmosphere: websocket.onerror
vaadinPush-min.js?v=23.2.4:1 Mon Oct 10 2022 15:16:01 GMT+0200 (heure d’été d’Europe centrale) Atmosphere: websocket.onclose
vaadinPush-min.js?v=23.2.4:1 Websocket closed, reason: Connection was closed abnormally (that is, with no close frame being sent). - wasClean: false
log @ vaadinPush-min.js?v=23.2.4:1
vaadinPush-min.js?v=23.2.4:1 Mon Oct 10 2022 15:16:01 GMT+0200 (heure d’été d’Europe centrale) Atmosphere: Firing onClose (closed case)
vaadinPush-min.js?v=23.2.4:1 Mon Oct 10 2022 15:16:01 GMT+0200 (heure d’été d’Europe centrale) Atmosphere: Request already closed, not firing onClose (closed case)
vaadinPush-min.js?v=23.2.4:1 Mon Oct 10 2022 15:16:01 GMT+0200 (heure d’été d’Europe centrale) Atmosphere: Firing onReconnect
vaadinPush-min.js?v=23.2.4:1 Mon Oct 10 2022 15:16:01 GMT+0200 (heure d’été d’Europe centrale) Atmosphere: Firing onReconnect
(index):1 [DOM] Input elements should have autocomplete attributes (suggested: "username"): (More info: https://goo.gl/9p2vKq) <input type=​"text" slot=​"input" id=​"input-vaadin-text-field-6" required name=​"username" autocorrect=​"off" aria-labelledby=​"label-vaadin-text-field-0">​   slot 
vaadinPush-min.js?v=23.2.4:1 Invoking executeWebSocket, using URL: wss://test.o-cell.fr/?v-r=push&v-uiId=0&v-pushId=5b3f1338-8707-4ef9-ba2c-d9055e81b859&X-Atmosphere-tracking-id=0&X-Atmosphere-Framework=3.1.2-javascript&X-Atmosphere-Transport=websocket&X-Atmosphere-TrackMessageSize=true&Content-Type=application/json; charset=UTF-8&X-atmo-protocol=true
vaadinPush-min.js?v=23.2.4:1 WebSocket connection to 'wss://test.o-cell.fr/?v-r=push&v-uiId=0&v-pushId=5b3f1338-8707-4ef9-ba2c-d9055e81b859&X-Atmosphere-tracking-id=0&X-Atmosphere-Framework=3.1.2-javascript&X-Atmosphere-Transport=websocket&X-Atmosphere-TrackMessageSize=true&Content-Type=application/json;%20charset=UTF-8&X-atmo-protocol=true' failed: 
ax @ vaadinPush-min.js?v=23.2.4:1
vaadinPush-min.js?v=23.2.4:1 Mon Oct 10 2022 15:16:06 GMT+0200 (heure d’été d’Europe centrale) Atmosphere: websocket.onerror
vaadinPush-min.js?v=23.2.4:1 Mon Oct 10 2022 15:16:06 GMT+0200 (heure d’été d’Europe centrale) Atmosphere: websocket.onclose
vaadinPush-min.js?v=23.2.4:1 Websocket closed, reason: Connection was closed abnormally (that is, with no close frame being sent). - wasClean: false
log @ vaadinPush-min.js?v=23.2.4:1
vaadinPush-min.js?v=23.2.4:1 Mon Oct 10 2022 15:16:06 GMT+0200 (heure d’été d’Europe centrale) Atmosphere: Request already closed, not firing onClose (closed case)
vaadinPush-min.js?v=23.2.4:1 Mon Oct 10 2022 15:16:06 GMT+0200 (heure d’été d’Europe centrale) Atmosphere: Request already closed, not firing onClose (closed case)
vaadinPush-min.js?v=23.2.4:1 Websocket failed on first connection attempt. Downgrading to long-polling and resending
logo_16.png:1          Failed to load resource: the server responded with a status of 404 ()
vaadinPush-min.js?v=23.2.4:1 Mon Oct 10 2022 15:16:13 GMT+0200 (heure d’été d’Europe centrale) Atmosphere: ajaxRequest.onreadystatechange, new state: 2
vaadinPush-min.js?v=23.2.4:1 Mon Oct 10 2022 15:16:13 GMT+0200 (heure d’été d’Europe centrale) Atmosphere: ajaxRequest.onreadystatechange, new state: 3
vaadinPush-min.js?v=23.2.4:1 Mon Oct 10 2022 15:16:13 GMT+0200 (heure d’été d’Europe centrale) Atmosphere: ajaxRequest.onreadystatechange, new state: 4
vaadinPush-min.js?v=23.2.4:1 Mon Oct 10 2022 15:16:13 GMT+0200 (heure d’été d’Europe centrale) Atmosphere: ajaxRequest.onreadystatechange, new state: 2
vaadinPush-min.js?v=23.2.4:1 Mon Oct 10 2022 15:16:13 GMT+0200 (heure d’été d’Europe centrale) Atmosphere: Firing onOpen
vaadinPush-min.js?v=23.2.4:1 Mon Oct 10 2022 15:16:13 GMT+0200 (heure d’été d’Europe centrale) Atmosphere: Firing onOpen
vaadinPush-min.js?v=23.2.4:1 Mon Oct 10 2022 15:16:13 GMT+0200 (heure d’été d’Europe centrale) Atmosphere: ajaxRequest.onreadystatechange, new state: 3
vaadinPush-min.js?v=23.2.4:1 Mon Oct 10 2022 15:16:13 GMT+0200 (heure d’été d’Europe centrale) Atmosphere: ajaxRequest.onreadystatechange, new state: 3
logo_16.png:1          Failed to load resource: the server responded with a status of 404 ()
FlowClient.638236c0.js:1 Uncaught (in promise) Error: Client is resynchronizing
    at FlowClient.638236c0.js:1:42744
    at Array.forEach (<anonymous>)
    at Zy (FlowClient.638236c0.js:1:42719)
    at by (FlowClient.638236c0.js:1:34159)
    at z.db (FlowClient.638236c0.js:3:76809)
    at x (FlowClient.638236c0.js:1:32329)
    at Map.forEach (<anonymous>)
    at h2 (FlowClient.638236c0.js:1:46370)
    at W1 (FlowClient.638236c0.js:3:50514)
    at mf (FlowClient.638236c0.js:3:37733)
    at z.Bb (FlowClient.638236c0.js:3:74767)
    at z.O (FlowClient.638236c0.js:3:86948)
    at XMLHttpRequest.<anonymous> (FlowClient.638236c0.js:1:25882)
    at od (FlowClient.638236c0.js:1:16760)
    at K2 (FlowClient.638236c0.js:3:5540)
    at XMLHttpRequest.<anonymous> (FlowClient.638236c0.js:1:28235)
`
This is a build which includes the flow 23.2.4 build released on friday.

**Comment by caalador** [comment]
This would seem to be some other issue, probably with push as it fails to connect a websocket connection.
Also some pwa icon requests seem to be made, perhaps there is an old service worker trying to fetch things?

If you could open up a new issue (as this one was about not recovering from resync) and if possible have a minimal application that would fail also it would help.

**Comment by echarlus** [comment]
@caalador thanks for the advice. New bug is here : [https://github.com/vaadin/flow/issues/14797](https://github.com/vaadin/flow/issues/14797)
FYI the 404 on the image is "normal" it's an image that should be displayed in my header and that is not always present, depending on the context.

**Comment by alexanoid** [comment]
Hi! I have the same issue with NGINX, Vaadin 23.2.4 - 

`2022-10-17 01:26:18.425  WARN 3555474 --- [http-nio-8080-exec-10] c.v.f.s.communication.ServerRpcHandler   : Resynchronizing UI by client's request. A network message was lost before reaching the client and the client is reloading the full UI state. This typically happens because of a bad network connection with packet loss or because of some part of the network infrastructure (load balancer, proxy) terminating a push (websocket or long-polling) connection. If you are using push with a proxy, make sure the push timeout is set to be smaller than the proxy connection timeout`

more details https://stackoverflow.com/questions/74081025/vaadin23-uncaught-in-promise-error-client-is-resynchronizing

**Comment by jonasrotilli** [comment]
This issue has not been resolved.
It keeps happening, even in the latest version 23.2.9.
Right now I'm seeing this happen, in production, internet ok, 1 hour with the screen open without moving, when changing screens, the problem started.

This is complicated, it's been 6 months since my report and it keeps happening, version comes in, version comes out and there is no solution or definitive explanation.

**Comment by echarlus** [comment]
I've also seen it a few times with 23.3.1 :(

**Comment by jonasrotilli** [comment]
I just updated to 23.2.9, and the finding is that it's much worse.
It happened 2 times already, just with me as I'm a system developer.

I will need to go back to the previous version.

> This issue has not been resolved. It keeps happening, even in the latest version 23.2.9. Right now I'm seeing this happen, in production, internet ok, 1 hour with the screen open without moving, when changing screens, the problem started.
> 
> This is complicated, it's been 6 months since my report and it keeps happening, version comes in, version comes out and there is no solution or definitive explanation.

**Comment by jonasrotilli** [comment]
This item needs to be re-opened, it has not been resolved.

**Comment by jonasrotilli** [comment]
See the browser log messages, when they happen:
.
.
.
.




Tue Jan 10 2023 16:00:58 GMT-0400 (Horário Padrão do Amazonas) Atmosphere: Firing onClose (closed case)
vaadinPush-min.js?v=23.2.3:1 Tue Jan 10 2023 16:00:58 GMT-0400 (Horário Padrão do Amazonas) Atmosphere: Request already closed, not firing onClose (closed case)

vaadinPush-min.js?v=23.2.3:1 Tue Jan 10 2023 16:00:58 GMT-0400 (Horário Padrão do Amazonas) Atmosphere: invoking .close() on WebSocket object

2vaadinPush-min.js?v=23.2.3:1 Tue Jan 10 2023 16:00:58 GMT-0400 (Horário Padrão do Amazonas) Atmosphere: Firing onReconnect

vaadinPush-min.js?v=23.2.3:1 Invoking executeWebSocket, using URL: wss://portal.upcampo.com.br/?v-r=push&v-uiId=0&v-pushId=c72c1567-11f6-4e9f-81ab-d51fd96ba37e&X-Atmosphere-tracking-id=725954af-eeb4-42ca-94a0-7905083d0fec&X-Atmosphere-Framework=3.1.2-javascript&X-Atmosphere-Transport=websocket&X-Atmosphere-TrackMessageSize=true&Content-Type=application/json; charset=UTF-8&X-atmo-protocol=true
vaadinPush-min.js?v=23.2.3:1 Tue Jan 10 2023 16:01:05 GMT-0400 (Horário Padrão do Amazonas) Atmosphere: websocket.onopen

vaadinPush-min.js?v=23.2.3:1 Websocket successfully opened
vaadinPush-min.js?v=23.2.3:1 Tue Jan 10 2023 16:01:05 GMT-0400 (Horário Padrão do Amazonas) Atmosphere: websocket.onmessage

2vaadinPush-min.js?v=23.2.3:1 Tue Jan 10 2023 16:01:05 GMT-0400 (Horário Padrão do Amazonas) Atmosphere: Firing onReopen

generated-flow-imports.81274417.js:894 WARNING: Since Vaadin 23, notifyResize() is deprecated. The component uses a ResizeObserver internally and doesn't need to be explicitly notified of resizes.

notifyResize @ generated-flow-imports.81274417.js:894
**11FlowClient.947c8d40.js:1 Uncaught (in promise) Error: Client is resynchronizing**
    at FlowClient.947c8d40.js:1:42744
    at Array.forEach (<anonymous>)
    at Zy (FlowClient.947c8d40.js:1:42719)
    at by (FlowClient.947c8d40.js:1:34159)
    at z.db (FlowClient.947c8d40.js:3:76660)
    at x (FlowClient.947c8d40.js:1:32329)
    at Map.forEach (<anonymous>)
    at i2 (FlowClient.947c8d40.js:1:43877)
    at W1 (FlowClient.947c8d40.js:3:50372)
    at mf (FlowClient.947c8d40.js:3:37591)
    at z.Bb (FlowClient.947c8d40.js:3:74625)
    at z.O (FlowClient.947c8d40.js:3:86799)
    at XMLHttpRequest.<anonymous> (FlowClient.947c8d40.js:1:25882)
    at od (FlowClient.947c8d40.js:1:16760)
    at Z2 (FlowClient.947c8d40.js:3:5692)
    at XMLHttpRequest.<anonymous> (FlowClient.947c8d40.js:1:28235)

**Comment by jonasrotilli** [comment]
Curious that it happens a lot more on Mac computers.
I tested it on more than one mac computer and it happened on both.
Already on a Windows, the problem occurs much less frequently.
I tested it in Google Chrome browser and Safari, in both the same behavior. If I open the browsers log, it shows an error: 

**(in promise) Error: Client is resynchronizing at FlowClient.42a5821f.js:1:42438**

We need an urgent solution for this, the situation is becoming unsustainable, many customers are complaining.

There are several tickects open for this and no solution:
#15205
#14797

I am ready to help you by providing information.
If anyone wants to connect to my computer and look at the problem happening, if they want my project, anyway, I'm available to help with the definitive resolution.

**Comment by knoobie** [comment]
If you can reproduce the problem locally, that would probably help a lot. 

Additionally you can enable trace logging to https://github.com/vaadin/flow/blob/master/flow-server/src/main/java/com/vaadin/flow/component/internal/UIInternals.java which would print all information why the server side count increases (A LOT) to find the cause why / how it happens. (This should not be added to a production env)

**Comment by mshabarov** [comment]
@jonasrotilli thanks for willing to help on this issue. Would you be up for a call to discuss and see what is happening on your side? Please email me mikhail@vaadin.com and we can then arrange a meet. We are working on the other issues related to client-server resynchronisation, but since this one is getting worse, we'll work on it asap.

**Comment by jonasrotilli** [comment]
> If you can reproduce the problem locally, that would probably help a lot.
> 
> Additionally you can enable trace logging to https://github.com/vaadin/flow/blob/master/flow-server/src/main/java/com/vaadin/flow/component/internal/UIInternals.java which would print all information why the server side count increases (A LOT) to find the cause why / how it happens. (This should not be added to a production env)

I don't have the problem locally, only when it's production.
What I meant is that we can make a call to view here on my computer occurring in production.
There was a time when the problem was random. Now it happens in a pattern: just let the system stand still for a few minutes, it happens.

**Comment by jonasrotilli** [comment]
> @jonasrotilli thanks for willing to help on this issue. Would you be up for a call to discuss and see what is happening on your side? Please email me [mikhail@vaadin.com](mailto:mikhail@vaadin.com) and we can then arrange a meet. We are working on the other issues related to client-server resynchronisation, but since this one is getting worse, we'll work on it asap.

I sent my data, so we can talk, thanks!

**Comment by jonasrotilli** [comment]
> Se você puder reproduzir o problema localmente, isso provavelmente ajudará muito.
> 
> Além disso, você pode ativar o registro de rastreamento em https://github.com/vaadin/flow/blob/master/flow-server/src/main/java/com/vaadin/flow/component/internal/UIInternals.java que imprimiria todos informações por que a contagem do lado do servidor aumenta (MUITO) para encontrar a causa / como isso acontece. (Isso não deve ser adicionado a um ambiente de produção)

Could you help me to put this TRACE?
Is there somewhere with an explanation how should I use it?

It would help me a lot, thanks.

**Comment by knoobie** [comment]
In Spring Boot for example add this to your application.yml (you probably already have the top level entries):

```yml

logging:
  level:
    com.vaadin.flow.component.internal.UIInternals: TRACE

```

**Comment by jonasrotilli** [comment]
> No Spring Boot, por exemplo, adicione isso ao seu application.yml (você provavelmente já possui as entradas de nível superior):
> 
> ```yaml
> logging:
>   level:
>     com.vaadin.flow.component.internal.UIInternals: TRACE
> ```

I don't use application.yml, I use application.properties, if you have the lines ready, help me!
I'm searching

**Comment by mshabarov** [comment]
@jonasrotilli 
this works for me 
add `logging.level.com.vaadin.flow.component.internal = trace` in application.properties

**Comment by jonasrotilli** [comment]
After help from @mshabarov  and  @mcollovati , they guided me to put version 23.2.13.
In this version the problem is gone.
No other changes were made, I didn't touch NGINX, anything in the application or anything related to the server, so I believe that something between version 23.2.9 and 23.2.13 solved the problem.

Thanks for the help!

I won't close, so that someone with more knowledge can assess what has been done between these versions.

**Comment by Artur-** [comment]
#14887 was fixed in 23.2.12 and 23.3.2. There does not seem to be any comments here about newer versions experiencing the problem

**Comment by mshabarov** [comment]
Closed because the issue is not reproduced for author. If anyone would experience the same issue again, please report it here, so we can re-open and investigate, or make a new issue in Flow.

**Comment by dbMe1** [comment]
Intermittently, in PROD environment only, I am seeing:
  UnsupportedOperationException: Unexpected message id from the client. Expected sync id: 3, got 4

JDK 17
Spring Boot 3.2.0
Vaadin 24.2.4

Please help.
Here is most of the stacktrace:

2023-12-17T04:58:32.360Z ERROR 7 --- [dbME] [nio-6386-exec-9] c.v.flow.server.DefaultErrorHandler      :

java.lang.UnsupportedOperationException: Unexpected message id from the client. Expected sync id: 3, got 4. more details logged on DEBUG level.

at com.vaadin.flow.server.communication.ServerRpcHandler.handleRpc(ServerRpcHandler.java:313) ~[flow-server-24.2.3.jar!/:24.2.3]

at com.vaadin.flow.server.communication.UidlRequestHandler.synchronizedHandleRequest(UidlRequestHandler.java:114) ~[flow-server-24.2.3.jar!/:24.2.3]

at com.vaadin.flow.server.SynchronizedRequestHandler.handleRequest(SynchronizedRequestHandler.java:40) ~[flow-server-24.2.3.jar!/:24.2.3]

at com.vaadin.flow.server.VaadinService.handleRequest(VaadinService.java:1522) ~[flow-server-24.2.3.jar!/:24.2.3]

at com.vaadin.flow.server.VaadinServlet.service(VaadinServlet.java:398) ~[flow-server-24.2.3.jar!/:24.2.3]

at com.vaadin.flow.spring.SpringServlet.service(SpringServlet.java:106) ~[vaadin-spring-24.2.3.jar!/:na]

at jakarta.servlet.http.HttpServlet.service(HttpServlet.java:658) ~[tomcat-embed-core-10.1.16.jar!/:na]

at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:205) ~[tomcat-embed-core-10.1.16.jar!/:na]

at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:149) ~[tomcat-embed-core-10.1.16.jar!/:na]

at org.apache.catalina.core.ApplicationDispatcher.invoke(ApplicationDispatcher.java:642) ~[tomcat-embed-core-10.1.16.jar!/:na]

at org.apache.catalina.core.ApplicationDispatcher.processRequest(ApplicationDispatcher.java:408) ~[tomcat-embed-core-10.1.16.jar!/:na]

at org.apache.catalina.core.ApplicationDispatcher.doForward(ApplicationDispatcher.java:313) ~[tomcat-embed-core-10.1.16.jar!/:na]

at org.apache.catalina.core.ApplicationDispatcher.forward(ApplicationDispatcher.java:277) ~[tomcat-embed-core-10.1.16.jar!/:na]

at org.springframework.web.servlet.mvc.ServletForwardingController.handleRequestInternal(ServletForwardingController.java:141) ~[spring-webmvc-6.1.1.jar!/:6.1.1]

at org.springframework.web.servlet.mvc.AbstractController.handleRequest(AbstractController.java:178) ~[spring-webmvc-6.1.1.jar!/:6.1.1]

at org.springframework.web.servlet.mvc.SimpleControllerHandlerAdapter.handle(SimpleControllerHandlerAdapter.java:51) ~[spring-webmvc-6.1.1.jar!/:6.1.1]

at org.springframework.web.servlet.DispatcherServlet.doDispatch(DispatcherServlet.java:1089) ~[spring-webmvc-6.1.1.jar!/:6.1.1]

at org.springframework.web.servlet.DispatcherServlet.doService(DispatcherServlet.java:979) ~[spring-webmvc-6.1.1.jar!/:6.1.1]

at org.springframework.web.servlet.FrameworkServlet.processRequest(FrameworkServlet.java:1014) ~[spring-webmvc-6.1.1.jar!/:6.1.1]

at org.springframework.web.servlet.FrameworkServlet.doPost(FrameworkServlet.java:914) ~[spring-webmvc-6.1.1.jar!/:6.1.1]

at jakarta.servlet.http.HttpServlet.service(HttpServlet.java:590) ~[tomcat-embed-core-10.1.16.jar!/:na]

at org.springframework.web.servlet.FrameworkServlet.service(FrameworkServlet.java:885) ~[spring-webmvc-6.1.1.jar!/:6.1.1]

at jakarta.servlet.http.HttpServlet.service(HttpServlet.java:658) ~[tomcat-embed-core-10.1.16.jar!/:na]

at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:205) ~[tomcat-embed-core-10.1.16.jar!/:na]

at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:149) ~[tomcat-embed-core-10.1.16.jar!/:na]

at org.apache.tomcat.websocket.server.WsFilter.doFilter(WsFilter.java:51) ~[tomcat-embed-websocket-10.1.16.jar!/:na]

at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:174) ~[tomcat-embed-core-10.1.16.jar!/:na]

at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:149) ~[tomcat-embed-core-10.1.16.jar!/:na]

at org.springframework.security.web.FilterChainProxy.lambda$doFilterInternal$3(FilterChainProxy.java:231) ~[spring-security-web-6.2.0.jar!/

**Comment by TetianaAntonova** [comment]
Hello guys! 
Please help me to fix this.
<img width="1668" alt="Image" src="https://github.com/user-attachments/assets/aacf2a16-6b03-4de6-b1ff-3b20953131c6" />
